ui/js/directives/clonejobs.js

```diff
@@ -81,21 +81,22 @@ treeherder.directive('thCloneJobs', [
             };
         }
 
-        jobs = $(selector);
-        var idx = jobs.index(jobs.filter(".selected-job"));
-        idx = getIndex(idx, jobs);
+        jobs = $(selector).filter(":visible");
+        if (jobs.length) {
+            var selIdx = jobs.index(jobs.filter(".selected-job"));
+            var idx = getIndex(selIdx, jobs);
 
-        el = $(jobs[idx]);
-        while (el.css('display') === 'none') {
-            idx = getIndex(idx, jobs);
             el = $(jobs[idx]);
+            key = el.attr(jobKeyAttr);
+            if (jobMap && jobMap[key] && selIdx !== idx) {
+                selectJob(jobMap[key].job_obj);
+                return;
+            }
         }
-
-        key = el.attr(jobKeyAttr);
-        if (jobMap && jobMap[key]) {
-            selectJob(jobMap[key].job_obj);
-        }
-
+        // if there was no new job selected, then ensure that we clear any job that
+        // was previously selected.
+        $timeout($rootScope.closeJob, 0);
```

Comment:
I notice another case also similar to the above. If we have a resultset (or I guess filtered view) with only 1 unclassified failure, typing n,p also closes the view of that unclassified failure and we get the new "No matching job select" warning.

I would expect we'd want to just behave like stage/prod and preserve that single unclassified selection and job details panel, with no warning.

![singleunclassified](https://cloud.githubusercontent.com/assets/3660661/8561242/868ebf70-24f0-11e5-8351-0472f28fd52c.jpg)